analyze_visual/wrapper.py

```diff
@@ -14,53 +14,72 @@
 import os.path
 from os import path
 import numpy as np
+from sklearn.metrics import accuracy_score
+from sklearn.model_selection import train_test_split
+from pickle import load
 sys.path.insert(0, '..')
-from analyze_visual.analyze_visual import process_video
+from train import feature_extraction, data_preparation
 
 def parse_arguments():
     """Parse arguments for real time demo.
     """
     parser = argparse.ArgumentParser(description="Wrapper"
                                                  "Predict shot class")
-    parser.add_argument("-f", "--file", required=True, nargs=None,
-                        help="File")
+
+    parser.add_argument("-v", "--videos_path", required=True, action='append',
+                        nargs='+', help="Videos folder path")
 
     parser.add_argument("-m", "--model", required=True, nargs=None,
                         help="Model")
 
     return parser.parse_args()
 
-def model_predict(features,model):
+def model_predict(features,labels,algorithm):
     """
     Loads pre-trained model and predict shot's class
     :param features: features
+    :labels: labels
+    :algorithm: Training algorithm
     :return:
     """
-    #Load pre-trained model
-    loaded_model = pickle.load(open('trained_'+str(model)+'.sav', 'rb'))
-    #Predict the class
-    result = loaded_model.predict(features)
 
-    print('This shot belongs to class: ', result)
+    _, X_test, _, y_test = train_test_split(features, labels, test_size=0.33)
+
+    # Load the model
+    model = load(open('trained_'+str(algorithm)+'.pkl', 'rb'))
+
+    # Load the scaler
+    scaler = load(open(str(algorithm)+'_scaler.pkl', 'rb'))
+
+    # Transform the test dataset
+    X_test_scaled = scaler.transform(X_test)
+
+    # Predict the class
+    results = model.predict(X_test_scaled)
+    
+    #Print the results
+    acc = accuracy_score(y_test, results)
+    print('Test Accuracy of classifier: ', acc)
+    print('The shots belongs to class: ', results)
 
 
 if __name__ == "__main__":
 
     args = parse_arguments()
-    shot = args.file
-    model = args.model
-    print(shot)
-
-    #Check if npy file of video file exists
-    if path.exists(str(shot)+'_shot_features.npy'):
-        features_stats = np.load(str(shot)+'_shot_features.npy')
-        features = features_stats.reshape(1,-1)
-    else:
-        features_stats = process_video(shot, 2, True, True, True)
-        features = features_stats[0]
-        features = features.reshape(1,-1)
-
-    model_predict(features,model)
+    videos_path = args.videos_path
+    algorithm = args.model
+
+    # Convert list of lists to a single list
+    videos_path = [item for sublist in videos_path for item in sublist]
```

Comment:
Please check if this is a file or folder. If folder do what u'r doing now. If file, just run the wrapper on a single video